File: 181 excersises/169.cpp
#include <iostream>
#include <vector>
using namespace std;

void imprimirZigZag(int filas, int columnas) {
    vector<vector<int>> matriz(filas, vector<int>(columnas));
    int valor = 1;
    bool haciaAbajo = false;

    for (int col = 0; col < columnas; col++) {
        if (haciaAbajo) {
            for (int fila = filas - 1; fila >= 0; fila--) {
                matriz[fila][col] = valor++;
            }
        } else {
            for (int fila = 0; fila < filas; fila++) {
                matriz[fila][col] = valor++;
            }
        }
        haciaAbajo = !haciaAbajo;
    }

    for (int fila = 0; fila < filas; fila++) {
        for (int col = 0; col < columnas; col++) {
            cout << matriz[fila][col] << "\t";
        }
        cout << endl;
    }
}

int main() {
    int filas, columnas;
    cout << "Ingresa numero de filas: ";
    cin >> filas;
    cout << "Ingresa numero de columnas: ";
    cin >> columnas;

    imprimirZigZag(filas, columnas);

    return 0;
}

File: 181 excersises/169.py
def imprimir_zig_zag(filas, columnas):
    matriz = [[0 for _ in range(columnas)] for _ in range(filas)]
    valor = 1
    hacia_abajo = False

    for col in range(columnas):
        if hacia_abajo:
            for fila in range(filas-1, -1, -1):
                matriz[fila][col] = valor
                valor += 1
        else:
            for fila in range(filas):
                matriz[fila][col] = valor
                valor += 1
        hacia_abajo = not hacia_abajo
    for fila in matriz:
        print("\t".join(map(str, fila)))

filas = int(input("Ingresa el numero de filas: "))
columnas = int(input("Ingresa el número de columnas: "))

imprimir_zig_zag(filas, columnas)